data.py

```diff
@@ -28,17 +28,37 @@ def retrieve_paginated_data(self, suffix=None):
                       f"Content: {page_response.text}\nData processed before the error returned")
         return data
 
-    def get_data(self, rec_id=None, output='df'):
+    def get_data(self, rec_id=None, name=None, unique_key_available=None, refresh=None, output='df'):
         """Retrieve list of data sources or retrieve the contents of a data source
         rec_id : int
             RecId of the data you want to retrieve
+        name : str
+            data set name based filter to apply
+        unique_key_available : bool
+            filter data sets based on whether they're using unique keys
+        refresh : bool
+            filter data sets based on whether they're refresh-able (non-static)
         output : str
             Output format, either df or dict
         """
         if rec_id is None:
-            # Check token for expiry
-            data = self.session.api_call(suffix='data', request_method='get')
-            # return data.json()
+            # Add parameter string based on input
+            params = {}
+            if type(name) == str:
+                params["name"] = name
+            if type(unique_key_available) == bool:
+                params["uniquekeyavailable"] = unique_key_available
+            if type(refresh) == bool:
+                params["refresh"] = refresh
+            if params:
+                # Use api_call to grab list of data sets
+                data = self.session.api_call(suffix='data',
+                                             request_method='get',
+                                             params=params)
+            else:
+                # Use api_call to grab list of data sets
+                data = self.session.api_call(suffix='data', request_method='get')

```

Comment:
I noticed that you have three if statements that initialize `params`... the only time when `params` will be empty is if all three if statements test the wrong type, furthermore, you do a check on `params` inside `Session.api_call()`. Having said this, I would remove the if statements, and just initialize `params` and call `data = self.session.api_call(suffix='data', request_method='get', params=params)`. This approach would follow the principle from my previous comment, which is to let `Session.api_call()` do the check closer to where `params` is actually being used.